src/families/cosmos/mock.js

```diff
@@ -0,0 +1,170 @@
+// @flow
+import Prando from "prando";
+import { BigNumber } from "bignumber.js";
+import type { Account } from "../../types";
+import type {
+  CosmosOperation,
+  CosmosDelegateTxInfo,
+  CosmosResources,
+  CosmosDelegation
+} from "./types";
+
+import preloadedData from "./preloadedData.mock";
+
+import { genHex, genAddress } from "../../mock/helpers";
+
+const { validators } = preloadedData;
+
+/**
+ * Generates a cosmos delegation operation updating both operations list and account cosmos resources
+ * @memberof cosmos/mock
+ * @param {Account} account
+ * @param {Prando} rng
+ */
+function genDelegationOperation(account: Account, rng: Prando): Account {
+  const { spendableBalance, operations: ops } = account;
+
+  if (spendableBalance.isZero()) return account;
+
+  const address = genAddress(account.currency, rng);
+
+  const value = spendableBalance;
+
+  /** select position on the operation stack where we will insert the new delegation */
+  const opIndex = rng.next(0, 10);
+
+  const lastOp = ops[opIndex];
+  const date = new Date(
+    (lastOp ? lastOp.date.valueOf() : Date.now()) -
+      rng.nextInt(0, 100000000 * rng.next() * rng.next())
+  );
+
+  /** select between 3 to 5 validators and split the amount evenly */
+  const delegatedValidators = Array(rng.nextInt(3, 5))
```

Comment:
This can be `Array.from({ length: rng.nextInt(3, 5) })` then remove `fill(null)`